src/main/java/edu/unc/lib/boxc/migration/cdm/services/AspaceRefIdService.java

```diff
@@ -0,0 +1,218 @@
+            }
+        }
+
+        setUpdatedDate(Instant.now());
+    }
+
+    /**
+     * Syncs aspace ref id mappings from the mapping file into the database. Clears out any previously synced
+     * aspace ref id mapping details before updating.
+     * @throws IOException
+     */
+    public void syncMappings() throws IOException {
+        assertProjectStateValid();
+        if (project.getProjectProperties().getAspaceRefIdMappingsUpdatedDate() == null
+                || Files.notExists(project.getAspaceRefIdMappingPath())) {
+            throw new InvalidProjectStateException("Project has not previously generated aspace ref id mappings");
+        }
+
+        Connection conn = null;
+        try {
+            conn = indexService.openDbConnection();
+            Statement stmt = conn.createStatement();
+            // Cleanup any previously synced grouping data
+            cleanupStaleSyncedRefIds(stmt);
+            if (project.getProjectProperties().getAspaceRefIdMappingsSyncedDate() != null) {
+                setSyncedDate(null);
+            }
+
+            // Sync the aspace ref ids into the database
+            AspaceRefIdInfo info = loadMappings();
+            for (AspaceRefIdInfo.AspaceRefIdMapping entry : info.getMappings()) {
+                String cdmId = entry.getCdmId();
+                String aspaceRefId = entry.getAspaceRefId();
+
+                updateAspaceRefIdEntry(stmt, cdmId, aspaceRefId);
+            }
+        } catch (SQLException e) {
+            throw new MigrationException("Error interacting with export index", e);
+        } finally {
+            CdmIndexService.closeDbConnection(conn);
+        }
+        setSyncedDate(Instant.now());
+    }
+
+    private void cleanupStaleSyncedRefIds(Statement stmt) throws SQLException {
+        // Clear out of date aspace ref ids
+        stmt.executeUpdate("update " + CdmIndexService.TB_NAME
+                + " set " + CdmIndexService.ASPACE_REF_ID + " = null ");
+    }
+
+    private void updateAspaceRefIdEntry(Statement stmt, String aspaceRefId, String cdmId) throws SQLException {
+        stmt.executeUpdate("update " + CdmIndexService.TB_NAME
+                + " set " + CdmIndexService.ASPACE_REF_ID + " = '" + aspaceRefId + "'"
+                + " where " + CdmFieldInfo.CDM_ID + " = '"  + cdmId + "'");
+    }
+
+    protected void setUpdatedDate(Instant timestamp) throws IOException {
+        project.getProjectProperties().setAspaceRefIdMappingsUpdatedDate(timestamp);
+        ProjectPropertiesSerialization.write(project);
+    }
+
+    private void setSyncedDate(Instant timestamp) throws IOException {
+        project.getProjectProperties().setAspaceRefIdMappingsSyncedDate(timestamp);
+        ProjectPropertiesSerialization.write(project);
+    }
+
+    public Path getMappingPath() {
+        return project.getAspaceRefIdMappingPath();
+    }
+
+    private List<String> getIds() {
+        if (ids == null) {
```

Comment:
I'd suggest not using an instance variable for `ids`, it means that the service can only be used to load ids once. If the method were going to be called a lot of times it'd make sense to cache the value, but i think it's only being used once, so it probably makes more sense to keep the service stateless